### PR TITLE
[web-console] Show a rendering progress bar over the ELK layout pass

### PR DESCRIPTION
**Pull request by Karakatiza666**
When a profile zip is downloaded fro a pipeline in the profiler layout, the download progress bar is displayed, but after it is downloaded there is synchronous step of computing the diagram layout, which had no progress indication. THis PR fixes this - uses the same progress bar to indicate the layout is loading. There is no text that describes what's happening.

Testing: manual, ~added unit test~

### CODE REVIEW

File: js-packages/profiler-lib/src/renderingProgress.test.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { describe, expect, it } from 'vitest'
+import type { ProfilerCallbacks } from './profiler.js'
+
+// `onRenderingChange` is the lifecycle hook the profiler UI consumes to drive its
+// "rendering" progress bar (issue: in-progress feedback during the ELK layout pass,
+// which dominates first-render wall time on large profiles). These tests pin the
+// contract — the callback is optional, is invoked with a boolean, and pairs `true`
+// (layout starts) with `false` (layout finishes).
+
+describe('ProfilerCallbacks.onRenderingChange contract', () => {
```
Comment (mihaibudiu) [inline js-packages/profiler-lib/src/renderingProgress.test.ts]: I am not sure how useful this test is. The real question is whether this works properly with cytograph's background layout engine.

**Review by mihaibudiu — APPROVED** [review]

**Review by mythical-fred — APPROVED** [review]
LGTM — clean callback wiring with good edge-case handling (try/catch on synchronous throw, `dispose()` cleanup for in-flight layouts). Indeterminate bar is honest UX given ELK has no incremental progress signal. Test covers the contract nicely.
